src/schema/fair.ts

```diff
@@ -244,21 +244,31 @@ export const FairType = new GraphQLObjectType<any, ResolverContext>({
         },
       }),
       resolve: ({ id }, options, { fairBoothsLoader }) => {
-        const gravityOptions = omit(
-          convertConnectionArgsToGravityArgs(options),
-          ["page"]
-        )
-        gravityOptions.sort = gravityOptions.sort || "-featured"
-
-        return Promise.all([
-          totalViaLoader(fairBoothsLoader, id, gravityOptions),
-          fairBoothsLoader(id, gravityOptions),
-        ]).then(([count, { body: { results } }]) => {
-          return connectionFromArraySlice(results, options, {
-            arrayLength: count,
-            sliceStart: gravityOptions.offset,
+        interface GravityOptions {
+          size: number
+          sort: string
+          cursor?: string
+          section: string
+        }
+        const gravityOptions: GravityOptions = {
+          sort: "-featured",
+          section: options.section,
+          size: options.first,
+        }
+        if (!!options.after) {
+          gravityOptions.cursor = options.after
+        }
+        return fairBoothsLoader(id, gravityOptions)
+          .then(({ body: { results, next } }) => {
+            const connection = connectionFromArraySlice(results, options, {
+              arrayLength: results.length,
+              sliceStart: 0,
+            })
+            connection.pageInfo.endCursor = next
+            connection.pageInfo.hasNextPage = !!next
+            return connection
           })
-        })
+          .catch(() => null)
```

Comment:
Did you run into an error that you needed to silence? This can be added specifically when you'd want to ignore `fairBoothsLoader` error'ing out, so unless you ran into something we can leave it out.

Reply:
typescript is requiring I add a catch

Reply:
Hmm, interesting! What do you mean by that? I pulled this branch down locally and removed it and didn't seem to cause any problems with linting or anything. Doing a `catch(null)` means real errors in that resolver won't get thrown/reported, so we don't want to do that unless there's a valid reason that a loader might throw an error that we want to allow to fail silently.

Reply:
Yeah, we don't want a silent error - perhaps moving this to an async function would de-neuter whatever was forcing you to add the catch (maybe tslint?)

Reply:
hm, strange, I swear it was requiring it, but removed it altogether and it's good now. Also added a test for the cursor logic under Ash's recommendation.